src/mir/optimizations.md

```diff
@@ -60,3 +60,24 @@ want to query whether your optimization pass should run, you can check the
 current level using `tcx.sess.opts.debugging_opts.mir_opt_level`.
 
 [compiler MCP]: https://github.com/rust-lang/compiler-team/issues/319
+
+## Optimization fuel
+
+Optimization fuel is a compiler option (`-Z fuel=<crate>=<value>`) that allows for fine grained
+control over which optimizations can be applied during compilation: each optimization reduces
+fuel by 1, and when fuel reaches 0 no more optimizations are applied. This can help with debugging
+and identifying problems with optimizations.
```

Comment:
Can you also leave a comment about this giving the ability to bisect optimizations by using a kind of binary search through the optimizations by using different fuels to home in on the problematic optimization? This explanation was the eye opener for me. I had no clue what fuel was for before your PR!

Reply:
I've added a small bit about that here. I hope that using terms like bisect and binary search make it clear how fuel can be used in this way...

Reply:
Thanks!